src/server/ui.js

```diff
@@ -88,3 +107,52 @@ export const getClientPaymentData = () => {
 
   return JSON.stringify(clientPaymentData);
 };
+
+export const addPaymentRecord = data => {
+  const {
+    paymentBreakdownData,
+    rate,
+    companyName,
+    initials,
+    datePaid,
+  } = JSON.parse(data);
+  const sheetValues = getPaymentSheetValues();
+  const newPaymentRecord = Array(sheetValues[0].length);
+  const paymentId = uuid();
+
+  newPaymentRecord[PAYMENT_ID_INDEX_ON_PAYMENT_OVERVIEW_SHEET] = paymentId;
+  newPaymentRecord[RATE_INDEX_ON_PAYMENT_OVERVIEW_SHEET] = rate;
+  newPaymentRecord[DATE_PAID_INDEX_ON_PAYMENT_OVERVIEW_SHEET] = getDate(
+    datePaid
+  );
+  newPaymentRecord[PAID_BY_INDEX_ON_PAYMENT_OVERVIEW_SHEET] = initials;
+  newPaymentRecord[COMPANY_NAME_INDEX_ON_PAYMENT_OVERVIEW_SHEET] = companyName;
+
+  const formattedPaymentBreakdownData = paymentBreakdownData.map(client => {
+    const { startDate, endDate, id, terminationDate } = client;
+    const paidThroughDate = getDate(
+      client[PAID_THROUGH_DATE_INDEX_ON_CLIENT_SHEET]
+    );
+    const eDate = getDate(endDate);
+    const sDate = getDate(startDate);
+    const tDate = terminationDate && getDate(terminationDate);
```

Comment:
Does this still work if there is no end date? Need to check but first have to be able to clear datepicker